exercises/custom-set/example.rs

```diff
@@ -1,149 +1,100 @@
-use std::iter::{IntoIterator, FromIterator};
-use std::slice;
-use std::option::Option;
-use std::vec::Vec;
-
 #[derive(Debug)]
 pub struct CustomSet<T> {
-    elements: Vec<T>
+    collection: Vec<T>,
 }
 
-/// Create an empty set.
-impl <T> CustomSet<T> where T: Eq + Clone {
-    pub fn new() -> CustomSet<T> {
-        CustomSet { elements: Vec::new() }
+impl<T: Ord + Clone> PartialEq for CustomSet<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.collection.iter().all(|x| other.contains(&x)) &&
+        other.collection.iter().all(|x| self.contains(&x))
     }
+}
 
-    pub fn len(&self) -> usize {
-        self.elements.len()
+impl<T: PartialEq + Ord + Clone> CustomSet<T> {
+    pub fn new(inputs: Vec<T>) -> CustomSet<T> {
+        let mut s = CustomSet { collection: Vec::new() };
+        for input in inputs {
+            s.add(input);
+        }
+        s
     }
 
-    pub fn is_empty(&self) -> bool {
-        self.len() == 0
+    pub fn add(&mut self, element: T) {
+        if !self.contains(&element) {
+            self.collection.push(element)
+        }
     }
 
-    pub fn contains(&self, needle: &T) -> bool {
-        for element in self.elements.iter() {
-            if needle == element {
-                return true;
-            }
-        }
-        false
+    pub fn delete(&mut self, element: T) {
```

Comment:
Well I note that HashSet sees no problem with that: [insert](https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.insert) takes an owned T, and [remove](https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.remove) borrows